base/logging.go

```diff
@@ -10,16 +10,95 @@
 package base
 
 import (
+	"errors"
 	"fmt"
+	"github.com/couchbase/clog"
+	"github.com/natefinch/lumberjack"
 	"log"
 	"os"
 	"runtime"
 	"strings"
 	"sync"
 	"time"
-	"github.com/couchbase/clog"
 )
 
+var errMarshalNilLevel = errors.New("can't marshal a nil *Level to text")
+
+type Level int32
+
+const (
+	// DebugLevel logs are typically voluminous, and are usually disabled in
+	// production.
+	DebugLevel Level = iota - 1
+	// InfoLevel is the default logging priority.
+	InfoLevel
+	// WarnLevel logs are more important than Info, but don't need individual
+	// human review.
+	WarnLevel
+	// ErrorLevel logs are high-priority. If an application is running smoothly,
+	// it shouldn't generate any error-level logs.
+	ErrorLevel
+	// PanicLevel logs a message, then panics.
+	PanicLevel
+	// FatalLevel logs a message, then calls os.Exit(1).
+	FatalLevel
+)
+
+// String returns a lower-case ASCII representation of the log level.
+func (l Level) String() string {
+	switch l {
+	case DebugLevel:
+		return "debug"
+	case InfoLevel:
+		return "info"
+	case WarnLevel:
+		return "warn"
+	case ErrorLevel:
+		return "error"
+	case PanicLevel:
+		return "panic"
+	case FatalLevel:
+		return "fatal"
+	default:
+		return fmt.Sprintf("Level(%d)", l)
+	}
+}
+
+// MarshalText marshals the Level to text. Note that the text representation
+// drops the -Level suffix (see example).
+func (l *Level) MarshalText() ([]byte, error) {
+	if l == nil {
+		return nil, errMarshalNilLevel
+	}
+	return []byte(l.String()), nil
+}
+
+// UnmarshalText unmarshals text to a level. Like MarshalText, UnmarshalText
+// expects the text representation of a Level to drop the -Level suffix (see
+// example).
+//
+// In particular, this makes it easy to configure logging levels using YAML,
+// TOML, or JSON files.
+func (l *Level) UnmarshalText(text []byte) error {
+	switch string(text) {
+	case "debug":
+		*l = DebugLevel
+	case "info":
+		*l = InfoLevel
+	case "warn":
+		*l = WarnLevel
+	case "error":
+		*l = ErrorLevel
+	case "panic":
+		*l = PanicLevel
+	case "fatal":
+		*l = FatalLevel
+	default:
+		return fmt.Errorf("unrecognized level: %v", string(text))
+	}
+	return nil
+}
+
 // 1 enables regular logs, 2 enables warnings, 3+ is nothing but panics.
 // Default value is 1.
 var logLevel int = 1
```

Comment:
The new Log levels are mapped to the existing internal SG logging levels as follows:

// DebugLevel	-1 --> 1
// InfoLevel 	 0 --> 1
// WarnLevel 	 1 --> 2
// ErrorLevel 	 2 --> 2
// PanicLevel 	 3 --> 3
// FatalLevel 	 4 --> 3